src/test/java/org/threeten/extra/chrono/TestInternationalFixedChronology.java

```diff
@@ -0,0 +1,1184 @@
+            {2012, -1, -1, ALIGNED_DAY_OF_WEEK_IN_MONTH, 7, 2012, 7, 7},
+
+            {2012, -1, -1, ALIGNED_DAY_OF_WEEK_IN_YEAR, 0, 2012, -1, -1},
+            {2012, -1, -1, ALIGNED_DAY_OF_WEEK_IN_YEAR, 1, 2012, 7, 1},
+            {2012, -1, -1, ALIGNED_DAY_OF_WEEK_IN_YEAR, 2, 2012, 7, 2},
+            {2012, -1, -1, ALIGNED_DAY_OF_WEEK_IN_YEAR, 3, 2012, 7, 3},
+            {2012, -1, -1, ALIGNED_DAY_OF_WEEK_IN_YEAR, 4, 2012, 7, 4},
+            {2012, -1, -1, ALIGNED_DAY_OF_WEEK_IN_YEAR, 5, 2012, 7, 5},
+            {2012, -1, -1, ALIGNED_DAY_OF_WEEK_IN_YEAR, 6, 2012, 7, 6},
+            {2012, -1, -1, ALIGNED_DAY_OF_WEEK_IN_YEAR, 7, 2012, 7, 7},
+
+            {2012, -1, -1, ALIGNED_WEEK_OF_MONTH, 0, 2012, -1, -1},
+            {2012, -1, -1, ALIGNED_WEEK_OF_MONTH, 3, 2012, 7, 15},
+
+            {2012, -1, -1, ALIGNED_WEEK_OF_YEAR, 0, 2012, -1, -1},
+            {2012, -1, -1, ALIGNED_WEEK_OF_YEAR, 3, 2012, 1, 15},
+
+            {2012, -1, -1, DAY_OF_WEEK, 0, 2012, -1, -1},
+            {2012, -1, -1, DAY_OF_WEEK, 1, 2012, 7, 1},
+            {2012, -1, -1, DAY_OF_WEEK, 2, 2012, 7, 2},
+            {2012, -1, -1, DAY_OF_WEEK, 3, 2012, 7, 3},
+            {2012, -1, -1, DAY_OF_WEEK, 4, 2012, 7, 4},
+            {2012, -1, -1, DAY_OF_WEEK, 5, 2012, 7, 5},
+            {2012, -1, -1, DAY_OF_WEEK, 6, 2012, 7, 6},
+            {2012, -1, -1, DAY_OF_WEEK, 7, 2012, 7, 7},
+
+            {2012, -1, -1, DAY_OF_MONTH, 0, 2012, 0, 0},
+            {2012, -1, -1, DAY_OF_MONTH, -1, 2012, -1, -1},
+            {2012, -1, -1, DAY_OF_MONTH, 3, 2012, 7, 3},
+
+            {2012, -1, -1, MONTH_OF_YEAR, -1, 2012, -1, -1},
+            {2012, -1, -1, MONTH_OF_YEAR, 0, 2012, 0, 0},
+            {2012, -1, -1, MONTH_OF_YEAR, 7, 2012, 7, 1},
+            {2012, -1, -1, MONTH_OF_YEAR, 2, 2012, 2, 1},
+
+            {2012, -1, -1, YEAR, 2012, 2012, -1, -1},
+            {2012, -1, -1, YEAR, 2013, 2013, 7, 1},
+            {2012, -1, -1, YEAR, 2011, 2011, 7, 1},
+            {2012, -1, -1, YEAR, 2016, 2016, -1, -1},
+
+            {2012, 3, 28, DAY_OF_MONTH, -1, 2012, -1, -1},
+            {2012, 1, 28, DAY_OF_MONTH, -1, 2012, -1, -1},
+            {2012, 3, 28, MONTH_OF_YEAR, -1, 2012, -1, -1},
+            {2012, 3, 28, DAY_OF_YEAR, 169, 2012, -1, -1},
+            {2013, 3, 28, DAY_OF_YEAR, 169, 2013, 7, 1},
+            {2013, 7, 1, YEAR, 2012, 2012, 7, 1},
+        };
+    }
+
+    @Test(dataProvider = "with")
+    public void test_with_TemporalField(int year, int month, int dom,
+            TemporalField field, long value,
+            int expectedYear, int expectedMonth, int expectedDom) {
+        assertEquals(InternationalFixedDate.of(year, month, dom).with(field, value), InternationalFixedDate.of(expectedYear, expectedMonth, expectedDom));
+    }
+
+    @DataProvider(name = "with_bad")
+    Object[][] data_with_bad() {
+        return new Object[][] {
+            {2013, 1, 1, DAY_OF_WEEK, -1},
```

Comment:
Use this data instead for `data_with_bad()`:

            {2013, 1, 1, ALIGNED_DAY_OF_WEEK_IN_MONTH, 0},
            {2013, 1, 1, ALIGNED_DAY_OF_WEEK_IN_MONTH, 8},
            {2012, 1, 1, ALIGNED_DAY_OF_WEEK_IN_MONTH, 0},
            {2012, 1, 1, ALIGNED_DAY_OF_WEEK_IN_MONTH, 8},

            {2013, 1, 1, ALIGNED_DAY_OF_WEEK_IN_YEAR, 0},
            {2013, 1, 1, ALIGNED_DAY_OF_WEEK_IN_YEAR, 8},
            {2012, 1, 1, ALIGNED_DAY_OF_WEEK_IN_YEAR, 0},
            {2012, 1, 1, ALIGNED_DAY_OF_WEEK_IN_YEAR, 8},

            {2013, 1, 1, ALIGNED_WEEK_OF_MONTH, 0},
            {2013, 1, 1, ALIGNED_WEEK_OF_MONTH, 5},
            {2012, 1, 1, ALIGNED_WEEK_OF_MONTH, 0},
            {2012, 1, 1, ALIGNED_WEEK_OF_MONTH, 5},

            {2013, 1, 1, ALIGNED_WEEK_OF_YEAR, 0},
            {2013, 1, 1, ALIGNED_WEEK_OF_YEAR, 53},
            {2012, 1, 1, ALIGNED_WEEK_OF_YEAR, 0},
            {2012, 1, 1, ALIGNED_WEEK_OF_YEAR, 53},

            {2013, 1, 1, DAY_OF_WEEK, 0},
            {2013, 1, 1, DAY_OF_WEEK, 8},
            {2012, 1, 1, DAY_OF_WEEK, 0},
            {2012, 1, 1, DAY_OF_WEEK, 8},

            {2013, 1, 1, DAY_OF_MONTH, -1},
            {2013, 1, 1, DAY_OF_MONTH, 29},
            {2012, 1, 1, DAY_OF_MONTH, -2},
            {2012, 1, 1, DAY_OF_MONTH, 29},

            {2013, 1, 1, DAY_OF_YEAR, 0},
            {2012, 1, 1, DAY_OF_YEAR, 0},
            {2013, 1, 1, DAY_OF_YEAR, 366},
            {2012, 1, 1, DAY_OF_YEAR, 367},

            {2013, 1, 1, MONTH_OF_YEAR, -1},
            {2013, 1, 1, MONTH_OF_YEAR, 14},
            {2012, 1, 1, MONTH_OF_YEAR, -2},
            {2012, 1, 1, MONTH_OF_YEAR, 14},

            {2013, 1, 1, YEAR, 0}

There were gaps in the data, that had effects on how you decided to implement.  Some of this is my fault, as I pulled from Discordian slightly too liberally.  `0` is now only valid if _currently on_ year-day/leap-day, and does nothing (returns same day).  Otherwise, it's an invalid value (you can't ask for the 0th day-of-week in most circumstances, after all).